tests/token_federation/github_oidc_test.py

```
@@ -0,0 +1,168 @@
+#!/usr/bin/env python3
+
+"""
+Test script for Databricks SQL token federation with GitHub Actions OIDC tokens.
+
+This script tests the Databricks SQL connector with token federation
+using a GitHub Actions OIDC token. It connects to a Databricks SQL warehouse,
+runs a simple query, and shows the connected user.
+"""
+
+import os
+import sys
+import json
+import base64
+import logging
+from databricks import sql
+
+
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s - %(levelname)s - %(message)s"
+)
+logger = logging.getLogger(__name__)
+
+
+def decode_jwt(token):
+    """
+    Decode and return the claims from a JWT token.
+    
+    Args:
+        token: The JWT token string
+        
+    Returns:
+        dict: The decoded token claims or None if decoding fails
+    """
+    try:
+        parts = token.split(".")
+        if len(parts) != 3:
+            raise ValueError("Invalid JWT format")
+        
+        payload = parts[1]
+        # Add padding if needed
+        padding = '=' * (4 - len(payload) % 4)
+        payload += padding
+        
+        decoded = base64.b64decode(payload)
+        return json.loads(decoded)
+    except Exception as e:
+        logger.error(f"Failed to decode token: {str(e)}")
+        return None
+
+
+def get_environment_variables():
+    """
+    Get required environment variables for the test.
+    
+    Returns:
+        tuple: (github_token, host, http_path, identity_federation_client_id)
+    
+    Raises:
+        SystemExit: If any required environment variable is missing
+    """
+    github_token = os.environ.get("OIDC_TOKEN")
+    if not github_token:
```

Feels like too much error handling. If some variable is not present let it fail by Null pointer exception, Code breaking exceptions don't think need to be handled

---

I feel this is okay, I wanted the CI/CD to be verbose. made a few changes but keeping the core.